test/test_tensorclass_nofuture.py

```diff
@@ -207,23 +221,34 @@ def test_batch_size():
 
 
 def test_len():
-    myc = MyData(X=torch.rand(2, 3, 4), y=torch.rand(2, 3, 4, 5), batch_size=[2, 3])
+    myc = MyData(
+        X=torch.rand(2, 3, 4),
+        y=torch.rand(2, 3, 4, 5),
+        z="test_tensorclass",
+        batch_size=[2, 3],
+    )
     assert len(myc) == 2
 
-    myc2 = MyData(X=torch.rand(2, 3, 4), y=torch.rand(2, 3, 4, 5), batch_size=[])
+    myc2 = MyData(
+        X=torch.rand(2, 3, 4),
+        y=torch.rand(2, 3, 4, 5),
+        z="test_tensorclass",
+        batch_size=[],
+    )
     assert len(myc2) == 0
 
 
 def test_indexing():
-    data = MyData(
-        X=torch.ones(3, 4, 5),
-        y=torch.zeros(3, 4, 5, dtype=torch.bool),
-        batch_size=[3, 4],
-    )
+    x = torch.ones(3, 4, 5)
+    y = torch.zeros(3, 4, 5, dtype=torch.bool)
+    z = ["a", "b", "c"]
+    data = MyData2(X=x, y=y, z=z, batch_size=[3, 4])
 
     assert data[:2].batch_size == torch.Size([2, 4])
     assert data[1].batch_size == torch.Size([4])
     assert data[1][1].batch_size == torch.Size([])
+    assert data[1].z == data[2].z == data[:2].z == z
+    assert data[:2].X.shape == torch.Size([2, 4, 5])
 
```

Comment:
we miss getitem/indexing test with a nested class.
@vmoens, it is not clear to me what should be the behaviour in this case

Reply:
Added nested case for indexing